apps/web/lib/hooks/useInViewObserver.ts

```diff
@@ -40,3 +40,42 @@ export const useInViewObserver = (onInViewCallback: () => void) => {
     ref: setRef,
   };
 };
+
+export const useInViewObserverWithNullRoot = (onInViewCallback: () => void) => {
+  const [node, setRef] = React.useState<HTMLElement | null>(null);
+
+  const onInViewCallbackRef = React.useRef(onInViewCallback);
+  onInViewCallbackRef.current = onInViewCallback;
```

Comment:
It could be just a flag in `useInViewObserver` right or better we could support a root argument in there